### PR TITLE
[root] add dataframe cmake option

### PR DESCRIPTION
**Pull request by vvolkl**
@chissg @HadrienG2  @drbenmorgan

This has been a separate cmake option starting v6-19 I believe: https://github.com/root-project/root/commit/31292b90826fbb1aaf0694c445f5996cdde3f358
It should default to true -- not sure why, but this recipe sets it to off.

I could add a variant too, but since it has become an integral part of root and doesn't introduce extra dependencies, I'd propose to just set it to true like I do here.

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
Waiting for approval from maintainers

**Comment by vvolkl** [comment]
@chissg @HadrienG2 @drbenmorgan If you are ok with this, I'm also adding myself to the maintainers list.

**Pull request merged**
